tests/test_tasker.py

```diff
@@ -403,3 +404,61 @@ def test_retry_method(retry_times):
             retry(my_func, *my_args, retry=retry_times, **my_kwargs)
     else:
         retry(my_func, *my_args, retry=retry_times, **my_kwargs)
+
+
+@pytest.mark.parametrize('exc', [ProtocolError, APIError, False])
+@pytest.mark.parametrize('in_init', [True, False])
+@pytest.mark.parametrize('retry_times', [-1, 0, 1, 2, 3])
+def test_retry_generator(exc, in_init, retry_times):
+    def simplegen():
+        yield "log line"
+
+    my_args = ('some', 'new')
+    if not in_init:
+        my_kwargs = {'one': 'first', 'two': 'second', 'retry_times': retry_times}
+    else:
+        my_kwargs = {'one': 'first', 'two': 'second'}
+
+    if in_init:
+        t = DockerTasker(retry_times=retry_times)
+    else:
+        t = DockerTasker()
+
+    (flexmock(time)
+        .should_receive('sleep')
+        .and_return(None))
+
+    if not exc:
+        cr = CommandResult()
+        cr._error = "cmd_error"
+        cr._error_detail = {"message": "error_detail"}
+
+    if exc == APIError:
+        error_message = 'api_error'
+        response = flexmock(content=error_message, status_code=408)
+        (flexmock(atomic_reactor.util)
+            .should_receive('wait_for_command')
+            .times(retry_times + 1)
+            .and_raise(APIError, error_message, response))
+    elif exc == ProtocolError:
+        error_message = 'protocol_error'
+        (flexmock(atomic_reactor.util)
+            .should_receive('wait_for_command')
+            .times(retry_times + 1)
+            .and_raise(ProtocolError, error_message))
+    else:
+        (flexmock(atomic_reactor.util)
+            .should_receive('wait_for_command')
+            .times(retry_times + 1)
+            .and_return(cr))
+        error_message = 'cmd_error'
+
+    if retry_times >= 0:
+        with pytest.raises(RetryGeneratorException) as exc:
+            t.retry_generator(lambda *args, **kwargs: simplegen(),
+                              *my_args, **my_kwargs)
+
+        assert repr(error_message) in exc.value.args[0]
```

Comment:
I still think we ought to be making an assertion about what `repr(exc)` looks like, since that's what the user will see, rather than rummaging inside `exc`, but I'm happy enough with this.

Reply:
but repr(exc) is this : <ExceptionInfo RetryGeneratorException tblen=2>
that isn't what user will see

Reply:
Oh, it's an ExceptionInfo (I thought it was an Exception because it is named `exc` not `exc_info`). Then `repr(exc.value)` is what ought to be tested.

The existing test verifies that the exception message contains the actual error detail somewhere, but doesn't verify that the resulting exception shown in a Koji task result will actually be intelligible to users.

Reply:
ok will use repr(exc.value)